src/api/MetadataQuery.js

```diff
@@ -0,0 +1,137 @@
+/**
+ * @flow
+ * @file Helper for the box metadata query API
+ * @author Box
+ */
+
+import Base from './Base';
+import { CACHE_PREFIX_METADATA, ERROR_CODE_METADATA_QUERY } from '../constants';
+
+class MetadataQuery extends Base {
+    /**
+     * @property {string}
+     */
+    key: string;
+
+    /**
+     * @property {Function}
+     */
+    successCallback: Function;
+
+    /**
+     * @property {Function}
+     */
+    errorCallback: ElementsErrorCallback;
+
+    /**
+     * Creates a key for the metadata cache
+     *
+     * @param {string} id - metadata template
+     * @return {string} key
+     */
+    getCacheKey(id: string): string {
+        return `${CACHE_PREFIX_METADATA}query_${id}`;
+    }
+
+    /**
+     * API URL for metadata query
+     * @return {string} base url for files
+     */
+    getUrl(): string {
+        return `${this.getBaseApiUrl()}/metadata_queries/execute`;
+    }
+
+    /**
+     * @param {Object} response
+     */
+    queryMetadataSuccessHandler = ({ data }: { data: Object }): void => {
+        const cache: APICache = this.getCache();
+        cache.set(this.key, data);
+        this.finish();
+    };
+
+    /**
+     * Does the network request to metadata query API
+     * @param {Object} query query object with SQL Clauses like properties
+     * @return {Promise}
+     */
+    queryMetadataRequest(query: Object): Promise<void> {
```

Comment:
Can both this and the success handler be grouped more closely with its caller, `queryMetadata`?